### PR TITLE
[shopsys] Added seo information to FEAPI

### PR DESCRIPTION
**Pull request by s3tezsky**
| Q             | A
| ------------- | ---
|Description, reason for the PR| SEO informations for products and categories were adde to frontend API.
|New feature| Yes 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/frontend-api/src/Model/Resolver/Category/CategoryResolverMap.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    /**
+     * @return array
+     */
+    protected function map(): array
+    {
+        return [
+            'Category' => [
+                'seoH1' => function (Category $category) {
```
Comment (grossmannmartin) [inline packages/frontend-api/src/Model/Resolver/Category/CategoryResolverMap.php:33]: Can you please consult with PO, whether we want to return the default value on null.

In administration you see a placeholder with default value (from general eshop seo settings).

Or there should be a query to return general seo settings.

**Review by grossmannmartin — CHANGES_REQUESTED** [review]

File: packages/frontend-api/src/Model/Resolver/Products/ProductResolverMap.php [hunk 2]
```diff
@@ -160,6 +160,18 @@ protected function mapProduct(): array
...
+            },
+            'seoTitle' => function ($data) {
+                return $data instanceof Product ? $data->getSeoTitle($this->domain->getId()) : $data['seo_title'];
+            },
+            'seoMetaDescription' => function ($data) {
+                if ($data instanceof Product) {
+                    return $data->getSeoMetaDescription($this->domain->getId());
+                }
+                return $data['seo_meta_description'];
+            },
```
[the remark this answers is not in the document]
Comment (s3tezsky) [reply packages/frontend-api/src/Model/Resolver/Products/ProductResolverMap.php:174]: Yes it is. I do not want to use such a long ternarry operator or format the ternar to multiline.

File: packages/framework/src/Model/Product/Elasticsearch/ProductExportRepository.php [hunk 3]
```diff
@@ -220,6 +220,9 @@ protected function extractResult(Product $product, int $domainId, string $locale
             'stock_quantity' => $product->getStockQuantity(),
             'variants' => $variantIds,
             'main_variant_id' => $product->isVariant() ? $product->getMainVariant()->getId() : null,
+            'seo_h1' => $product->getSeoH1($domainId),
+            'seo_title' => $product->getSeoTitle($domainId),
+            'seo_meta_description' => $product->getSeoMetaDescription($domainId),
```
[the remark this answers is not in the document]
Comment (s3tezsky) [reply packages/framework/src/Model/Product/Elasticsearch/ProductExportRepository.php:225]: It is reasonable, I will create another PR which should be merged first.

File: packages/frontend-api/src/Model/Resolver/Category/CategoryResolverMap.php (hunk 1 above)
Comment (sslt) [reply packages/frontend-api/src/Model/Resolver/Category/CategoryResolverMap.php:33]: I am more inclined to returning "null" value. It's up to developer of consumer App to decide if there should be default value and what it shoulld be.
...and yes, there should be query for returning "domain" data (discussed elsewhere), including generic SEO data.

File: packages/framework/src/Model/Product/Elasticsearch/ProductExportRepository.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (s3tezsky) [reply packages/framework/src/Model/Product/Elasticsearch/ProductExportRepository.php]: I have changed base to branch, where is this implemented.

File: packages/frontend-api/src/Model/Resolver/Products/ProductResolverMap.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (grossmannmartin) [reply packages/frontend-api/src/Model/Resolver/Products/ProductResolverMap.php:336]: okay

**Review by grossmannmartin — APPROVED** [review]

**Review by sslt — APPROVED** [review]

**Pull request merged**